target/arm/cpu.h

```diff
@@ -227,8 +241,24 @@ typedef struct CPUARMState {
      * instead of having a complicated union.
      */
     /* Regs for A64 mode.  */
+#ifdef TARGET_CHERI
+    struct GPCapRegs gpcapregs;
+    /* There was a choice here as to whether users of DDC should be indexing DDCs,
+     * or we should cache the current one and change it on exception. SP is swapped
+     * on exception, and doing so requires fewer global TCG vars, so is what we are doing
+     * here. */
+    cap_register_t DDC_current;
+    // Holds DDCs 0 through 3 then restricted 0
+    cap_register_t DDCs[5];
```

Comment:
Maybe like this?
```suggestion
    cap_register_t DDC_EL[4];
    cap_register_t RDDC_EL0;
```

Reply:
Keeping it as a single array makes it indexable. The SP bank mirrors this one, so having a single helper the returns an appropriate index for the current mode is neat.